src/components/container/ConnectDevicesMenu/ConnectDevicesMenu.tsx

```diff
@@ -59,7 +76,19 @@ export default function (props: ConnectDevicesMenuProps) {
                 } else {
                     setPlatform("Web");
                 }
-                setLoading(false);
+
+                if (deviceInfo?.platform == "iOS") {
+                    MyDataHelps.queryDeviceData({
+                        namespace: "AppleHealth",
+                        observedAfter: formatISO(add(new Date(), { days: -3 })),
+                        limit: 1
+                    }).then((result) => {
+                        setHasRecentAppleHealthData(result.deviceDataPoints.length > 0);
+                        setLoading(false);
+                    });
+                } else {
+                    setLoading(false);
+                }
             });
     }
 
```

Comment:
Existing code, but just a reminder that the below `useEffect` can be simplified to the following:

`useInitializeView(initialize, ["externalAccountSyncComplete"]);`